sharding/nodesSetup.go

```diff
@@ -102,14 +97,29 @@ func NewNodesSetup(
 		addressPubkeyConverter:   addressPubkeyConverter,
 		validatorPubkeyConverter: validatorPubkeyConverter,
 		genesisMaxNumShards:      genesisMaxNumShards,
+		genesisChainParameters:   chainParametersProvider.ChainParametersForEpoch(0),
 	}
 
-	err := core.LoadJsonFile(nodes, nodesFilePath)
-	if err != nil {
-		return nil, err
+	initialNodes := make([]*InitialNode, 0, len(nodesSetupDTO.InitialNodes))
+	for _, item := range nodesSetupDTO.InitialNodes {
```

Comment:
I needed to redeclare the InitialNode so I can place it into the `config` package. Also, the `InitialNode` from `sharding` package also includes the `nodeInfo` unexported item. So, I would say this is needed in order to avoid cyclic imports and keep the usage of the unexported `nodeInfo` item